### PR TITLE
Ajax lib

### PR DESCRIPTION
**Pull request by ajax-marchenko-s**
- add ajax nats lib
- add ajax kafka lib

### CODE REVIEW

File: domainservice/src/main/kotlin/ua/marchenko/artauction/artwork/controller/nats/GetAllArtworksFullNatsController.kt [hunk 1]
```diff
@@ -1,46 +1,49 @@
...
-    override val connection: Connection
-) : NatsController<FindAllArtworksFullRequestProto, FindAllArtworksFullResponseProto> {
+) : ProtoNatsMessageHandler<FindAllArtworksFullRequestProto, FindAllArtworksFullResponseProto> {
+
+    override val log: Logger = LoggerFactory.getLogger(GetAllArtworksFullNatsController::class.java)
 
     override val subject: String = NatsSubject.Artwork.FIND_ALL_FULL
 
-    override val queueGroup: String = QUEUE_GROUP
+    override val queue: String = QUEUE_GROUP
```
Comment (ajax-torbin-v) [inline domainservice/src/main/kotlin/ua/marchenko/artauction/artwork/controller/nats/GetAllArtworksFullNatsController.kt:26]: Very strange name for queue

File: domainservice/src/main/kotlin/ua/marchenko/artauction/auction/service/kafka/AuctionCreatedEventKafkaConsumer.kt [hunk 2]
```diff
@@ -1,40 +1,40 @@
...
+            .doOnSuccess { kafkaEvent.ack() }
     }
 
     private fun processAuctionEvent(event: AuctionCreatedEvent) {
         log.info("Received event: $event")
     }
 
     companion object {
-        private val log = LoggerFactory.getLogger(AuctionCreatedEventKafkaConsumer::class.java)
+        private const val CREATED_GROUP_ID = "auctionCreatedConsumerGroup"
```
Comment (ajax-torbin-v) [inline domainservice/src/main/kotlin/ua/marchenko/artauction/auction/service/kafka/AuctionCreatedEventKafkaConsumer.kt:38]: Why CREATED_GROUP_  _ID_ ?

File: domainservice/build.gradle.kts [hunk 3]
```diff
@@ -16,6 +16,9 @@ dependencies {
     implementation("io.nats:jnats:2.20.2")
```
Comment (ajax-shulepa-a) [inline domainservice/build.gradle.kts:16]: remove explicit jnats dependency

File: domainservice/src/main/kotlin/ua/marchenko/artauction/auction/service/kafka/AuctionCreatedEventNatsKafkaConsumer.kt [hunk 4]
```diff
@@ -1,52 +1,35 @@
...
-    private fun processAuctionEvent(event: AuctionCreatedEvent) {
-        natsConnection.publish(
-            NatsSubject.Auction.CREATED_EVENT,
-            event.auction.toAuctionProto(clock).toByteArray()
-        )
+    override fun handle(kafkaEvent: KafkaEvent<AuctionCreatedEventProto>): Mono<Unit> {
+        return natsPublisher.publish(
+            subject = NatsSubject.Auction.CREATED_EVENT,
+            payload = kafkaEvent.data.auction,
+        ).doOnSuccess { kafkaEvent.ack() }
```
Comment (ajax-shulepa-a) [inline domainservice/src/main/kotlin/ua/marchenko/artauction/auction/service/kafka/AuctionCreatedEventNatsKafkaConsumer.kt:29]: why did you change ack from doFinally to doOnSuccess?

**Review by ajax-shulepa-a — APPROVED** [review]

File: domainservice/src/main/kotlin/ua/marchenko/artauction/auction/service/kafka/AuctionCreatedEventNatsKafkaConsumer.kt (hunk 4 above)
Comment (ajax-marchenko-s) [reply domainservice/src/main/kotlin/ua/marchenko/artauction/auction/service/kafka/AuctionCreatedEventNatsKafkaConsumer.kt:29]: Write like in docs(
So here in my case there would be at least once semantics, because we ack only after success process

Comment (ajax-marchenko-s) [reply domainservice/src/main/kotlin/ua/marchenko/artauction/auction/service/kafka/AuctionCreatedEventNatsKafkaConsumer.kt]: changed to doFinally, because it is better for my business logic

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -40,12 +46,12 @@ jobs:
         run: docker compose --profile gh-actions up -d
 
       - name: Run gradle build
-        run: ./gradlew build --parallel
+        run: ${gradle_cmd} build --parallel
```
Comment (ajax-gnatiuk-s) [inline .github/workflows/build.yml:49]:
```suggestion
        run: ${gradle_cmd} build
```

File: domainservice/src/main/kotlin/ua/marchenko/artauction/artwork/controller/nats/AddArtworkNatsController.kt [hunk 6]
```diff
@@ -1,46 +1,49 @@
...
             .onErrorResume {
-                log.error("Error in CreateArtwork for {}", request, it)
+                log.error("Error in CreateArtwork for {}", inMsg, it)
                 it.toCreateArtworkFailureResponseProto().toMono()
             }
     }
 
     companion object {
-        private const val QUEUE_GROUP = "artwork"
-        private val log = LoggerFactory.getLogger(AddArtworkNatsController::class.java)
```
Comment (ajax-gnatiuk-s) [inline domainservice/src/main/kotlin/ua/marchenko/artauction/artwork/controller/nats/AddArtworkNatsController.kt:44]: declare logger in companion object. please, revert the change

Comment (ajax-gnatiuk-s) [reply domainservice/src/main/kotlin/ua/marchenko/artauction/artwork/controller/nats/AddArtworkNatsController.kt:44]: review all places

File: .github/workflows/build.yml (hunk 5 above)
Comment (ajax-marchenko-s) [reply .github/workflows/build.yml:49]: I have merged this pr, so I would add these changes in my open pr with Redis

File: domainservice/src/main/kotlin/ua/marchenko/artauction/artwork/controller/nats/AddArtworkNatsController.kt (hunk 6 above)
Comment (ajax-marchenko-s) [reply domainservice/src/main/kotlin/ua/marchenko/artauction/artwork/controller/nats/AddArtworkNatsController.kt:44]: I can't do it because here we are overriding logger
